### DISCUSSION TITLE
RuntimeError  Variable "dienstleistung" does not exist. at Symfony 6

### DISCUSSION THREAD

**Discussion by Linuxmaker** [symfony/symfony#50677] [Q&A] [answered]
Hello,
I have a controller here under Symfony 6.3 and PHP 8.1. Creating the record in the table works. Querying the table fails.
In the web browser I get these error messages instead of the output in the index file
Variable "service" does not exist.
AbstractController->render() in src/Controller/ServicesController.php (line 29)
The Controller:
```
<?php

namespace App\Controller;

use App\Entity\Dienstleistung;
use Doctrine\Persistence\ManagerRegistry;
use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
use Symfony\Component\HttpFoundation\Response;
use Symfony\Component\Routing\Annotation\Route;

class DienstleistungenController extends AbstractController
{
    #[Route('/dienstleistungen', name: 'app_dienstleistungen')]
    public function index(ManagerRegistry $doctrine): Response
    {
        $dienstleistung = new Dienstleistung;
        $dienstleistung -> setName('Systemadministration');
        
        //Entity-Manager
        $entityManager = $doctrine->getManager();

        //$entityManager->persist($dienstleistung);
        //$entityManager->flush();

        $getdienstleistung = $entityManager->getRepository(Dienstleistung::class)->findOneBy(['id' => 1]);
       
        //return new Response("Dienstleistung wurde angelegt!");
        return $this->render('dienstleistungen/index.html.twig', [
            'name' => $getdienstleistung
        ]);
    }
}
```
And this is the associated Twig file:
```
{% extends 'base.html.twig' %}

{% block title %}Hello DienstleistungenController!{% endblock %}

{% block body %}

{{dienstleistung.name}}

{% endblock %}
```
Here is the content of the mySQL-Table:
```
MariaDB [honorar]> SELECT * FROM dienstleistung;
+----+----------------------+----------+---------+
| id | name                 | beratung | support |
+----+----------------------+----------+---------+
|  1 | Systemadministration |     NULL |    NULL |
+----+----------------------+----------+---------+
1 row in set (0,000 sec)
```
I use VSCodium as an editor with Autocomplication and I've noticed that at this point
$getservice = $entityManager->getRepository(service::class)->f
the find commands like 'findOneBy' are not included.
The tutorial is designed for Symfony 5.4 and maybe it doesn't work that way in the current Symfony and nothing is assigned to 'name'.
At least I'm at a loss now and don't know how to render the column value from 'name' of the data set with id=1.
Maybe someone can help me here?
Thanks very much
Andreas

**Comment by ghost** [comment, marked as answer]
The error message is different in the discussion title and the description. Please proofread before you submit new discussion. 

The problem is here:

```php
        return $this->render('dienstleistungen/index.html.twig', [
            'name' => $getdienstleistung // you create a Twig variable called "name"
        ]);
```

```twig
{{dienstleistung.name}} # should be {{ name.name }}
```

**Reply by lyrixx** [reply]
@javaDeveloperKid  Good catch 👍🏼 

@Linuxmaker And I guess there was an exception embed in another exception. Try to scroll down to find the root cause of the issue: it'll display the error in the template

**Reply by Linuxmaker** [reply]
@javaDeveloperKid Thanks for your feedback. I'm just learning Symfony, so your answer is helpful.
